### PR TITLE
[BH-1635] Low battery notification on the home screen

### PR DESCRIPTION
**Pull request by ghost**
A low battery warning window has been added to the main screen. This notification will appear only once when the battery level falls to 15%. It will also appear when the battery level falls below 10% and 5%.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

File: products/BellHybrid/apps/common/src/BatteryLevelNotificationModel.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    {
+        auto updateBatteryIntervalState = [&](BatteryLevelInterval newBatteryLevelIntervalState) {
+            if (currentBatteryLevelInterval == newBatteryLevelIntervalState) {
+                return;
+            }
+            currentBatteryLevelInterval     = newBatteryLevelIntervalState;
+            isCurrentBatteryIntervalHandled = (newBatteryLevelIntervalState == BatteryLevelInterval::ABOVE_15_PERCENT);
+        };
+
+        auto updateBatteryIntervalStateWithHysteresis = [&](std::uint32_t above15PercentHysteresis,
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/common/src/BatteryLevelNotificationModel.cpp:21]: Consider removing last parameter as it is always equal to zero

File: products/BellHybrid/apps/common/src/UserSessionModel.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2017-2024, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "models/UserSessionModel.hpp"
+#include "log/log.hpp" //TODO: is it still needed?
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/common/src/UserSessionModel.cpp:5]: It is, as you're logging from inside this file :wink: But please include it with angle brackets

File: products/BellHybrid/apps/common/src/UserSessionModel.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        endSessionDelayTimer = sys::TimerFactory::createPeriodicTimer(
+            serv, "EndOfUserSessionDelay", endOfUserSessionTimeout, [this](sys::Timer &) {
+                endOfSessionTimerCallback();
+            });
+    }
+
+    void UserSessionModel::activateUserSession() // TODO: maybe rename to: refresh... or activeOrRefresh...
+    {
+        if (sessionState == SessionState::INACTIVE) {
+            LOG_INFO("User session begun");
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/common/src/UserSessionModel.cpp:21]: `began` or `has begun`

File: products/BellHybrid/apps/common/src/UserSessionModel.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            isCurrentUserSessionHandled = false;
+        }
+        sessionState = SessionState::ACTIVE;
+        endSessionDelayTimer.stop();
+    }
+
+    void UserSessionModel::restartUserSession()
+    {
+        LOG_INFO("User session begun (restarted)");
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/common/src/UserSessionModel.cpp:30]: Same as above :wink:

File: products/BellHybrid/apps/application-bell-main/windows/BellBatteryStatusWindow.hpp [hunk 5]
```diff
@@ -17,18 +17,20 @@ namespace gui
       public:
         struct Data : public gui::SwitchData
         {
-            Data(std::uint32_t lvl, bool chargeState) : chargeLevel{lvl}, isCharging{chargeState}
+            Data(std::uint32_t lvl, bool chargeState, bool lowBatteryWaring = false)
+                : chargeLevel{lvl}, isCharging{chargeState}, isItLowBatteryWaring{lowBatteryWaring}
             {}
             const std::uint32_t chargeLevel;
             const bool isCharging;
+            const bool isItLowBatteryWaring;
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/application-bell-main/windows/BellBatteryStatusWindow.hpp:25]:
```suggestion
            const bool isLowBatteryWarning;
```

File: products/BellHybrid/apps/application-bell-main/windows/BellBatteryStatusWindow.cpp [hunk 6]
```diff
@@ -113,10 +112,20 @@ namespace gui
         if (data != nullptr) {
             const auto &switchData = static_cast<Data &>(*data);
             const auto image       = battery_utils::getBatteryLevelImage(batteryEntries, switchData.chargeLevel);
-            if (image) {
-                center->setImage(image->data(), imageType);
+            if (switchData.isItLowBatteryWaring) {
+                topDescription->setText(utils::translate("battery_low"));
+                std::string tempBotDesc = utils::translate("battery_remaining");
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/application-bell-main/windows/BellBatteryStatusWindow.cpp:117]: I'd completely get rid of this variable and just put `utils::translate("battery_remaining")` directly in the function call

File: products/BellHybrid/apps/application-bell-main/windows/BellBatteryStatusWindow.hpp (hunk 5 above)
Comment (Maciej-Mudita) [inline products/BellHybrid/apps/application-bell-main/windows/BellBatteryStatusWindow.hpp:25]:
```suggestion
            const bool isItLowBatteryWarning;
```

File: products/BellHybrid/apps/common/include/common/models/BatteryLevelNotificationModel.hpp [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        BELOW_10_PERCENT, // 5-9%
+        BELOW_5_PERCENT,  // 0-4%
+    };
+
+    class AbstractBatteryLevelNotificationModel
+    {
+      public:
+        virtual ~AbstractBatteryLevelNotificationModel() noexcept = default;
+
+        virtual void updateBatteryLevelInterval(std::uint32_t currentBatteryLevel) = 0;
```
Comment (Maciej-Mudita) [inline products/BellHybrid/apps/common/include/common/models/BatteryLevelNotificationModel.hpp:40]: please consider using "units::SOC"

**Review by Maciej-Mudita — APPROVED** [review]

File: products/BellHybrid/apps/application-bell-main/presenters/StateController.cpp [hunk 8]
```diff
@@ -65,9 +65,15 @@ namespace app::home_screen
                 return true;
             };
 
-            auto switchToMenu          = [](AbstractPresenter &presenter) { presenter.switchToMenu(); };
-            auto switchToBatteryStatus = [](AbstractPresenter &presenter) { presenter.switchToBatteryStatus(); };
-            auto updateTemperature     = [](AbstractView &view, AbstractTemperatureModel &temperatureModel) {
+            auto switchToMenu             = [](AbstractPresenter &presenter) { presenter.switchToMenu(); };
+            auto isLowBatteryWaringToShow = [](AbstractPresenter &presenter) {
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/application-bell-main/presenters/StateController.cpp:69]:
```suggestion
            auto isLowBatteryWarningToShow = [](AbstractPresenter &presenter) {
```

File: products/PurePhone/services/evtmgr/WorkerEvent.cpp [hunk 9]
```diff
@@ -79,7 +79,7 @@ bool WorkerEvent::handleMessage(std::uint32_t queueID)
         if (notification == bsp::cellular::statusPin) {
             auto GSMstatus = bsp::cellular::status::getStatus();
             LOG_DEBUG("GSM Status pin change: %s",
-                      (GSMstatus == bsp::cellular::status::value::ACTIVE ? "ACTIVE" : "INACTIVE"));
+                      (GSMstatus == bsp::cellular::status::value::ACTIVE ? "ACTIVE" : "Inactive"));
```
Comment (Lefucjusz) [inline products/PurePhone/services/evtmgr/WorkerEvent.cpp:82]: Please "de-capitalize" both logs :wink:

**Review by Lefucjusz — APPROVED** [review]

**Pull request merged**